apps/api/src/modules/job/job.service.ts

```diff
@@ -235,8 +239,16 @@ export class JobService {
     return jobDetail
   }
 
-  update(id: number, updateJobDto: EditJobDto) {
-    return `This action updates a #${id} job`
+  async update(id: number, updateJobDto: EditJobDto, userId: number) {
+    const job = await this.repository.getJobById(id)
+    if (job.castingId !== userId) {
+      throw new ForbiddenException(
+        "You don't have permission to update this job",
+      )
+    }
+
+    this.validateJobDto(updateJobDto)
+    return this.repository.updateJob(id, updateJobDto, userId)
```

Comment:
this service (createJob too) should return in 
```
{
   jobId: number
}
```
or read swagger or read above controller for which type to return